src/app/functions/server/getBreaches.ts

```diff
@@ -5,40 +5,28 @@
 import { logger } from "./logging";
 import {
   HibpLikeDbBreach,
-  formatDataClassesArray,
   getAllBreachesFromDb,
-  req,
-} from "../../../utils/hibp.js";
+  fetchHibpBreaches,
+} from "../../../utils/hibp";
 import { upsertBreaches } from "../../../db/tables/breaches.js";
-import { Breach } from "../universal/breach.js";
 
-let breaches: Array<Breach | HibpLikeDbBreach>;
+let breaches: Array<HibpLikeDbBreach>;
 
-export async function getBreaches() {
+export async function getBreaches(): Promise<HibpLikeDbBreach[]> {
   if (breaches) {
     return breaches;
   }
   breaches = await getAllBreachesFromDb();
-  logger.debug(
-    "loadBreachesIntoApp",
-    `loaded breaches from database: ${breaches.length}`,
-  );
+  logger.debug(`loaded breaches from database: ${breaches.length}`);
 
   // if "breaches" table does not return results, fall back to HIBP request
   if (breaches?.length < 1) {
-    const breachesResponse = (await req("/breaches")) as Breach[];
-    logger.debug(
-      "loadBreachesIntoApp",
-      `loaded breaches from HIBP: ${breachesResponse.length}`,
-    );
-
-    for (const breach of breachesResponse) {
-      breach.DataClasses = formatDataClassesArray(breach.DataClasses);
-      breaches.push(breach);
-    }
+    const breachesResponse = await fetchHibpBreaches();
+    logger.debug(`loaded breaches from HIBP: ${breachesResponse.length}`);
 
     // sync the "breaches" table with the latest from HIBP
-    await upsertBreaches(breaches);
+    await upsertBreaches(breachesResponse);
+    breaches = await getAllBreachesFromDb();
```

Comment:
This is one of the few changes to runtime behaviour - instead of `breaches` having the data we just fetched from the HIBP API, it's the data we inserted into the database, which had a couple of transformations applied to it.

This should not cause issues, since we _already_ first tried to fetch the data from the database. Only if no breaches had been loaded in the database yet, did we send a request to the HIBP API and use its return value.